src/Symfony/Bundle/FrameworkBundle/Command/TranslationDebugCommand.php

```diff
@@ -86,134 +92,140 @@ protected function configure()
-        if (null !== $domain) {
-            $allMessages = array($domain => $allMessages);
-        }
+            // Extract used messages
+            $extractedCatalogue = $this->extractMessages($locale, $rootPath);
 
-        // No defined or extracted messages
-        if (empty($allMessages) || null !== $domain && empty($allMessages[$domain])) {
-            $outputMessage = sprintf('<info>No defined or extracted messages for locale "%s"</info>', $locale);
+            // Load defined messages
+            $currentCatalogue = $this->loadCurrentMessages($locale, $translationsPath, $loader);
 
+            // Merge defined and extracted messages to get all message ids
+            $mergeOperation = new MergeOperation($extractedCatalogue, $currentCatalogue);
+            $allMessages    = $mergeOperation->getResult()->all($domain);
             if (null !== $domain) {
-                $outputMessage .= sprintf(' <info>and domain "%s"</info>', $domain);
+                $allMessages = array($domain => $allMessages);
             }
 
-            $output->writeln($outputMessage);
-
-            return;
-        }
+            // No defined or extracted messages
+            if (empty($allMessages) || null !== $domain && empty($allMessages[$domain])) {
+                $outputMessage = sprintf('<info>No defined or extracted messages for locale "%s"</info>', $locale);
 
-        // Load the fallback catalogues
-        $fallbackCatalogues = array();
-        $translator = $this->getContainer()->get('translator');
-        if ($translator instanceof Translator) {
-            foreach ($translator->getFallbackLocales() as $fallbackLocale) {
-                if ($fallbackLocale === $locale) {
-                    continue;
+                if (null !== $domain) {
+                    $outputMessage .= sprintf(' <info>and domain "%s"</info>', $domain);
                 }
 
-                $fallbackCatalogue = new MessageCatalogue($fallbackLocale);
-                $loader->loadMessages($translationsPath, $fallbackCatalogue);
-                $fallbackCatalogues[] = $fallbackCatalogue;
+                $output->writeln($outputMessage);
+
+                continue;
             }
-        }
 
-        /** @var \Symfony\Component\Console\Helper\Table $table */
-        $table = new Table($output);
+            // Load the fallback catalogues
+            $fallbackCatalogues = $this->loadFallbackCatalogues($locale, $translationsPath, $loader);
 
-        // Display header line
-        $headers = array('State(s)', 'Domain', 'Id', sprintf('Message Preview (%s)', $locale));
-        foreach ($fallbackCatalogues as $fallbackCatalogue) {
-            $headers[] = sprintf('Fallback Message Preview (%s)', $fallbackCatalogue->getLocale());
-        }
-        $table->setHeaders($headers);
+            /** @var \Symfony\Component\Console\Helper\Table $table */
+            $table = new Table($output);
```

Comment:
@sgehrig sorry, I think this change not related to this PR, can you use the old.
 if we want to fix it, it should be in the old version.

I guess this would be my last comment so do not forget to squash your commits into one.